pkg/runtime/k0s/init.go

```diff
@@ -0,0 +1,80 @@
+// Copyright © 2022 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k0s
+
+import (
+	"fmt"
+)
+
+func (k *Runtime) init() error {
+	pipeline := []func() error{
+		k.MergeConfigOnMaster0,
+		k.SSHKeyGenAndCopyIDToHosts,
+		k.GenerateCert,
```

Comment:
This part of configuring private image repositories will be removed from the runtime pkg after we refactor run process, and i think k8s runtime also need those logic. see PR #1621

Reply:
Got it! will adapt it when new code merged.

Reply:
I suggest that we do this before the code merging. Or at least you add a TODO comment 
```
// TODO: move all these registry operation to the specific registry packages.
```